types/mapcmd/map_it_test.go

```diff
@@ -0,0 +1,448 @@
+package mapcmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/it"
+)
+
+// todo add value-file test
+func TestMapPut(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		mapNameArg := fmt.Sprintf(` --name %s`, mn)
+		tcs := []struct {
+			name        string
+			args        string
+			key         interface{}
+			value       interface{}
+			errContains string
+		}{
+			{
+				name:  "valid put key(string), value(string)",
+				args:  "--key k1 --value v1" + mapNameArg,
+				key:   "k1",
+				value: "v1",
+			},
+			{
+				name:  "valid put key(json), value(json)",
+				args:  `--key {"some":"key"} --key-type json --value {"some":"value"} --value-type json` + mapNameArg,
+				key:   serialization.JSON(`{"some":"key"}`),
+				value: serialization.JSON(`{"some":"value"}`),
+			},
+			{
+				name:        "invalid put, missing key",
+				args:        "--value v1" + mapNameArg,
+				errContains: `"key" not set`,
+			},
+			{
+				name:        "invalid put, missing map name",
+				args:        "--key k1 --value v1",
+				errContains: `"name" not set`,
+			},
+		}
+		for _, tc := range tcs {
+			t.Run(tc.name, func(t *testing.T) {
+				cmd := NewPut(c)
+				var stdout, stderr bytes.Buffer
+				cmd.SetOut(&stdout)
+				cmd.SetErr(&stderr)
+				args := strings.Split(tc.args, " ")
+				cmd.SetArgs(args)
+				ctx := context.Background()
+				_, err := cmd.ExecuteContextC(ctx)
+				if tc.errContains != "" {
+					require.Contains(t, err.Error(), tc.errContains)
+					return
+				}
+				require.NoError(t, err)
+				// cmd should not print anything
+				require.Empty(t, stdout.String())
+				require.Empty(t, stderr.String())
+				value, err := m.Get(ctx, tc.key)
+				require.NoError(t, err)
+				require.Equal(t, tc.value, value)
+			})
+		}
+	})
+}
+
+func TestMapPutAll_JSONEntries(t *testing.T) {
+	it.MapTesterWithConfigAndMapName(t, func(t *testing.T, c *hazelcast.Config, m *hazelcast.Map, mn string) {
+		f, err := ioutil.TempFile("", "entries.json")
```

Comment:
How about using `t.TempDIr()` instead of `ioutil.TempFile`?

Reply:
I didn't know that utility. Many thanks! https://github.com/hazelcast/hazelcast-commandline-client/pull/114/commits/33e5d0a48d545a710b929043f671ab0808485bcf